controller/src/main/java/io/pravega/controller/store/stream/BucketStore.java

```diff
@@ -82,7 +82,8 @@
 
     static int getBucket(String scope, String stream, int bucketCount) {
         String scopedStreamName = getScopedStreamName(scope, stream);
-        return scopedStreamName.hashCode() % bucketCount;
+        int hash = scopedStreamName.hashCode();
+        return hash == Integer.MIN_VALUE ? 0 : Math.abs(hash) % bucketCount;
```

Comment:
What about this:

```
return (hash ^ 0x7FFFFFFF) % bucketCount;
```

Reply:
If the purpose is to yield a positive number, then what you are suggesting looks better. It's faster. But if he needs a true `abs`, then what you are proposing will not account for 2's complement (though, IMO, I think just a positive number is good enough).

Reply:
> return (hash ^ 0x7FFFFFFF) % bucketCount;
I could use `&` instead of `^` (xor). We just need the sign bit to be flipped.

Reply:
I actually did mean to say or instead of xor, I'm glad you caught the it.